### PR TITLE
NXP17610: enable multiple file attachments

### PR DESCRIPTION
**Pull request by michaelva**

### CODE REVIEW

File: nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java [hunk 1]
```diff
@@ -212,14 +212,14 @@ protected Object resolveComplex(Element el, AttributeConfigDescriptor conf) {
         return propValue;
     }
 
-    protected Blob resolveBlob(Element el, AttributeConfigDescriptor conf) {
+    protected Blob resolveBlob(Element el, AttributeConfigDescriptor conf, String BlobPropertyName) {
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java]: `BlobPropertyName` -> `blobPropertyName`

File: nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Lesser General Public License
+ * (LGPL) version 2.1 which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/lgpl-2.1.html
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java]: LGPL -> APL

File: nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Test
+    public void test() throws Exception {
+
+        File xml = FileUtils.getResourceFileFromContext("multifiles.xml");
+        Assert.assertNotNull(xml);
+
+        DocumentModel root = session.getRootDocument();
+
+        XMLImporterService importer = Framework.getLocalService(XMLImporterService.class);
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java]: `getService`

File: nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Assert.assertNotNull(importer);
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java]: Use static import for `assertNotNull` etc.

File: nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+        session.save();
+
+        List<DocumentModel> docs = session.query("select * from Document where dc:title='Multifile'");
+        Assert.assertEquals("we should have only one File", 1, docs.size());
+
+        DocumentModel doc = docs.get(0);
+
+        Blob mainFile = (Blob) doc.getPropertyValue("file:content");
+        Assert.assertEquals("file1.txt",mainFile.getFilename());
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java]: Space after comma.

File: nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java [hunk 6]
```diff
@@ -302,8 +302,13 @@ protected void processDocAttributes(DocumentModel doc, Element el, AttributeConf
                     }
                 }
             } else {
-                value = (Serializable) resolveComplex(el, conf);
-                if (value != null) {
+                Map<String, Object> props = (Map<String, Object>) resolveComplex(el, conf);
+                value = new HashMap<>(props);
+                if (props.containsKey("file")) {
+                    Blob blob = resolveBlob(el, conf, "file");
+                    props.put("file",blob);
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java]: Space after comma.

File: nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+        List<Blob> attachments = (List<Blob>) doc.getPropertyValue("files:files");
+        Assert.assertEquals(2,attachments.size());
```
Comment (efge) [inline nuxeo-importer-xml-parser/src/test/java/org/nuxeo/ecm/platform/importer/xml/parser/test/TestMapperServiceMultiFiles.java]: Space after comma.

File: nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java [hunk 8]
```diff
@@ -263,7 +263,7 @@ protected void processDocAttributes(DocumentModel doc, Element el, AttributeConf
         } else if (property.isComplex()) {
 
             if (property instanceof BlobProperty) {
-                Object value = resolveBlob(el, conf);
+                Object value = resolveBlob(el, conf, "content");
```
Comment (andreinechaev) [inline nuxeo-importer-xml-parser/src/main/java/org/nuxeo/ecm/platform/importer/xml/parser/XMLImporterServiceImpl.java]: Isn't blobPropertyName == "content"?

**Pull request closed without merging**